pkg/asset/installconfig/openstack/openstack.go

```diff
@@ -236,7 +267,34 @@ func Platform() (*openstack.Platform, error) {
 		"OPENSHIFT_INSTALL_OPENSTACK_EXTERNAL_NETWORK",
 	)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to Marshal %s platform", openstack.Name)
+		return nil, err
+	}
+
+	flavorNames, err := getFlavorNames(cloud)
+	if err != nil {
+		return nil, err
+	}
+	flavor, err := asset.GenerateUserProvidedAsset(
+		"OpenStack Compute Flavor",
+		&survey.Question{
+			Prompt: &survey.Select{
+				Message: "FlavorName",
+				Help:    "The OpenStack compute flavor to use for servers. A flavor with at least 4 GB RAM is recommended.",
+				Options: flavorNames,
+			},
+			Validate: survey.ComposeValidators(survey.Required, func(ans interface{}) error {
+				value := ans.(string)
+				i := sort.SearchStrings(flavorNames, value)
+				if i == len(flavorNames) || flavorNames[i] != value {
+					return errors.Errorf("invalid flavor name %q, should be one of %+v", value, strings.Join(flavorNames, ", "))
+				}
+				return nil
+			}),
+		},
+		"OPENSHIFT_INSTALL_OPENSTACK_COMPUTE_FLAVOR",
```

Comment:
This is going to conflict with https://github.com/openshift/installer/pull/861. We are removing environment variables.

Reply:
OK, I can fix this up once that lands.

As for this being a prompt or not, the difference between AWS and OpenStack is whether it's possible to pick a sane default.  With AWS, we know what the flavors are, but for OpenStack, it's not standardized at all.